TheInternet/HoversPage.cs

```diff
@@ -0,0 +1,51 @@
+using OpenQA.Selenium;
+using OpenQA.Selenium.Interactions;
+
+namespace TheInternet;
+
+public class HoversPage : BasePage
+{
+    IList<IWebElement> userImagesList => driver.FindElements(By.CssSelector("div [class = 'figure']"));
+   
+    public HoversPage(IWebDriver driver) : base(driver)
+    {
+        this.driver = driver;
+    }
+
+    public UserCaption hoverOveruserImage(int userNumber)
+    {
+        userNumber = userNumber - 1;
+        Actions actions = new Actions(driver);
+        IWebElement user = userImagesList.ElementAt(userNumber);
+
+        actions.MoveToElement(user).Perform();
+        return new UserCaption(user.FindElement(By.ClassName("figcaption")));
+    }
+
+    public class UserCaption
+    {
+        private By username = By.TagName("h5");
```

Comment:
Convention:

If you look at the usage of this on line 38 (`return caption.Displayed;`) it's not always obviously clear in bigger methods whether `caption` is a local variable or a field (the term for a private variable within the class).

It also gets a bit muddy in the constructor where you use `this.` to disambiguate.

As a result, .NET convention is to name private fields like this one and `userLink` with an underscore prefix (e.g. `_userLink`)